clock_exercise/ActivationAlarmSwitch.qml

```diff
@@ -0,0 +1,55 @@
+import QtQuick 2.0
+
+Rectangle {
+    id: root
+
+    property bool on: false
+
+    signal changed
+
+    width: 112
+    height: 60
+    radius: 8
+    color: "#339FAAB0"
+
+    Image {
+        x: 70
+        y: 17
+        width: 26
+        height: 26
+        source: "images/223-bell.svg"
+        opacity: 0.3
+    }
+
+    Rectangle {
+        x: 30
+        y: 18
```

Comment:
Ho aspettato un po' prima di commentarti questo aspetto perchè non è fondamentale.
Va benissimo usare il posizionamento assoluto tramite X e Y, però cerchiamo anche di fare in modo che dal sorgente QML si riesca a visualizzare la resa e la disposizione degli elementi a schermo.
Con questo voglio dire che leggendo una serie di coordinate X e Y faccio un po' di fatica a capire come gli elementi sono disposti in relazione gli uni agli altri.

Ad esempio se due o più elementi sono uno sopra l'altro ad una certa distanza, posso valutare l'uso di una Column.
Discorso analogo per il caso orizzontale con la Row.

Se un elemento si trova spostato in basso a destra di una schermata, potrei ad esempio preferire vederlo posizionato in un modo simile piuttosto che usando `x:560; y: 940`:
```qml
Item {
    anchors.bottom: root.bottom
    anchros.right: root.right
    anchors.bottomMargin: 20
    anchors.rightMargin: 5
}
```

Decidere in quali casi usare posizionamento assoluto, piuttosto che usare gli ancoraggi, piuttosto che usare Column/Row è decisamente soggettivo, quindi non ti so indicare una regola programmatica da seguire, il consiglio che posso darti però è di cercare di individuare delle relazioni di posizionamento tra gli elementi che trovi nelle schermate e riportare quella stessa logica in QML.
Usare solo x e y chiaramente non aiuta il lettore a farsi un'idea delle relazioni di posizionamento tra oggetti.

Non serve che tu ti metta a cambiare tutto adesso, volevo solo commentarti questo aspetto.
Se vuoi possiamo approfondirlo meglio in riunione più tardi. :)

Reply:
Dopo aver fatto un po' di progetti ti dico che secondo me se hai un visual design abbastanza preciso con i posizionamenti spesso conviene seguire quello ed evitare accorpamenti arbitrari. Soprattutto in presenza di animazioni complesse usare le x,y assolute ti salva la vita

Reply:
Capisco bene che fai un discorso generico, ma per chiarezza ti rispondo considerando questo caso particolare, questo rectangle che hai segnalato non si capisce perché in realtà dovrebbe essere un immagine, mancava l'immagine per fare lo switch off e l'ho fatta io mettendo un rettangolino stretto.

![image](https://user-images.githubusercontent.com/70155823/96727848-cf203400-13b3-11eb-9fa3-9aa7570393b2.png)